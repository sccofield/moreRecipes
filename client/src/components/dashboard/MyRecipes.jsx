import React from 'react';
import ReactPaginate from 'react-paginate';
import moment from 'moment';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';

const MyRecipes = props => (
  <div>
    {props.userRecipes.length === 0 ? (
      <div>
        <p className="noRecipe">You have not created any recipe yet.</p>
        <p><Link to="/postRecipe">Click here</Link> to add a new recipe</p>
      </div>
    ) : (
      <section className="section text-center text-lg-left container">
        <nav className="row justify-content-center">
          <ReactPaginate
            previousLabel="previous"
            nextLabel="next"
            breakLabel={<a>...</a>}
            breakClassName="break-me"
            pageCount={props.pageCount}
            marginPagesDisplayed={2}
            pageRangeDisplayed={5}
            onPageChange={props.handlePageClick}
            containerClassName="pagination"
            subContainerClassName="pages pagination"
            activeClassName="active"
            pageClassName="page-item"
            pageLinkClassName="page-link"
            nextLinkClassName="page-link"
            previousLinkClassName="page-link"
          />
        </nav>

        {props.userRecipes.map(userRecipe => (
          <div key={userRecipe.id}>
            <div className="row">
              <div className="col-lg-4 mb-4">
                <div className="view overlay hm-white-slight z-depth-1">
                  <img
                    alt=""
                    src={userRecipe.imageUrl}
                    className="profileImage"
                  />
                  <a>
                    <div className="mask" />
                  </a>
                </div>
              </div>

              <div className="col-lg-7 tab-recipe">
                <h4 className="">
                  <Link to={`recipes/${userRecipe.id}`}>
                    <strong>{userRecipe.title}</strong>
                  </Link>
                </h4>
                <p>{userRecipe.description.slice(0, 80).concat(' ...')}</p>
                <p>
                  by
                  <a>
                    <strong> {userRecipe.userName}</strong>
                  </a>, { moment(userRecipe.createdAt).format('ll')}
                  <span className="btn btn-outline-info mx-1" id="editRecipe">
                    <Link className="" to={`/recipe/edit/${userRecipe.id}`}>
                      <i className="fa fa-pencil" aria-hidden="true" />
                    </Link>
                  </span>
                  <a
                    id="deleteRecipe"
                    className="btn btn-outline-info mx-1"
                    data-toggle="modal"
                    data-target="#deleteButton"
                  >
                    <i className="fa fa-trash" aria-hidden="true" />
                  </a>
                </p>

                {/* <-- Modal --> */}
                <div
                  className="modal fade"
                  id="deleteButton"
                  tabIndex="-1"
                  role="dialog"
                  aria-labelledby="exampleModalCenterTitle"
                  aria-hidden="true"
                >
                  <div
                    className="modal-dialog modal-dialog-centered"
                    role="document"
                  >
                    <div className="modal-content">
                      <div className="modal-header">
                        <h5 className="modal-title" id="exampleModalLongTitle">
                          Delete Recipe
                        </h5>
                        <button
                          type="button"
                          className="close"
                          data-dismiss="modal"
                          aria-label="Close"
                        >
                          <span aria-hidden="true">&times;</span>
                        </button>
                      </div>
                      <div className="modal-body">
                        <p>Are you sure you want to delete this recipe?</p>
                      </div>
                      <div className="modal-footer">
                        <button
                          type="button"
                          className="btn btn-secondary"
                          data-dismiss="modal"
                        >
                          No
                        </button>
                        <button
                          type="button"
                          className="btn btn-primary yesDelete"
                          onClick={props.onDelete}
                          id={userRecipe.id}
                        >
                          Yes
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>

            <hr />
          </div>
        ))}
        <p><Link to="/postRecipe">Click here</Link> to add a new recipe</p>

      </section>
    )}
  </div>
);

MyRecipes.propTypes = {
  userRecipes: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDelete: PropTypes.func.isRequired,
  pageCount: PropTypes.number,
  handlePageClick: PropTypes.func.isRequired
};
MyRecipes.defaultProps = {
  pageCount: null
};

export default MyRecipes;
