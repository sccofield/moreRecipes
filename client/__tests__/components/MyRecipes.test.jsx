import React from 'react';
import toJSON from 'enzyme-to-json';
import MyRecipes from '../../src/components/dashboard/MyRecipes';

describe('My Recipe component', () => {
  const props = {
    onDelete: jest.fn(() => Promise.resolve()),
    handlePageClick: jest.fn(() => Promise.resolve()),
    pageCount: 1,
    userRecipes: [{
      id: 3,
      title: 'title',
      imageUrl: 'image',
      description: 'description',
      userName: 'username',
      createdAt: 'createdat'


    }]
  };

  const noUserRecipesProps = {
    onDelete: jest.fn(() => Promise.resolve()),
    handlePageClick: jest.fn(() => Promise.resolve()),
    pageCount: 1,
    userRecipes: []
  };


  it('should render MyRecipesRecipes form when props is given', () => {
    const wrapper = shallow(<MyRecipes {...props} />);
    expect(wrapper.find('div').length).toEqual(13);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render text when props is not given', () => {
    const wrapper = shallow(<MyRecipes {...noUserRecipesProps} />);
    expect(wrapper.find('p').length).toEqual(2);
    expect(wrapper.find('p .noRecipe')
      .text()).toBe('You have not created any recipe yet.');
  });
});

