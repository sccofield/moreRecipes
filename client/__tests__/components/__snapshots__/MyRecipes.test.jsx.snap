// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Recipe component should render MyRecipesRecipes form when props is given 1`] = `
<div>
  <section
    className="section text-center text-lg-left container"
  >
    <nav
      className="row justify-content-center"
    >
      <PaginationBoxView
        activeClassName="active"
        breakClassName="break-me"
        breakLabel={
          <a>
            ...
          </a>
        }
        containerClassName="pagination"
        disableInitialCallback={false}
        disabledClassName="disabled"
        marginPagesDisplayed={2}
        nextClassName="next"
        nextLabel="next"
        nextLinkClassName="page-link"
        onPageChange={[MockFunction]}
        pageClassName="page-item"
        pageCount={1}
        pageLinkClassName="page-link"
        pageRangeDisplayed={5}
        previousClassName="previous"
        previousLabel="previous"
        previousLinkClassName="page-link"
        subContainerClassName="pages pagination"
      />
    </nav>
    <div
      key="3"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-4 mb-4"
        >
          <div
            className="view overlay hm-white-slight z-depth-1"
          >
            <img
              alt=""
              className="profileImage"
              src="image"
            />
            <a>
              <div
                className="mask"
              />
            </a>
          </div>
        </div>
        <div
          className="col-lg-7 tab-recipe"
        >
          <h4
            className=""
          >
            <Link
              replace={false}
              to="recipes/3"
            >
              <strong>
                title
              </strong>
            </Link>
          </h4>
          <p>
            description ...
          </p>
          <p>
            by
            <a>
              <strong>
                 
                username
              </strong>
            </a>
            , 
            Invalid date
            <span
              className="btn btn-outline-info mx-1"
              id="editRecipe"
            >
              <Link
                className=""
                replace={false}
                to="/recipe/edit/3"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-pencil"
                />
              </Link>
            </span>
            <a
              className="btn btn-outline-info mx-1"
              data-target="#deleteButton"
              data-toggle="modal"
              id="deleteRecipe"
            >
              <i
                aria-hidden="true"
                className="fa fa-trash"
              />
            </a>
          </p>
          <div
            aria-hidden="true"
            aria-labelledby="exampleModalCenterTitle"
            className="modal fade"
            id="deleteButton"
            role="dialog"
            tabIndex="-1"
          >
            <div
              className="modal-dialog modal-dialog-centered"
              role="document"
            >
              <div
                className="modal-content"
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                    id="exampleModalLongTitle"
                  >
                    Delete Recipe
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      Ã—
                    </span>
                  </button>
                </div>
                <div
                  className="modal-body"
                >
                  <p>
                    Are you sure you want to delete this recipe?
                  </p>
                </div>
                <div
                  className="modal-footer"
                >
                  <button
                    className="btn btn-secondary"
                    data-dismiss="modal"
                    type="button"
                  >
                    No
                  </button>
                  <button
                    className="btn btn-primary yesDelete"
                    id={3}
                    onClick={[MockFunction]}
                    type="button"
                  >
                    Yes
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <p>
      <Link
        replace={false}
        to="/postRecipe"
      >
        Click here
      </Link>
       to add a new recipe
    </p>
  </section>
</div>
`;
